lib/screen/clubs/clubs_screen.dart

```diff
@@ -51,15 +51,15 @@ class ClubsScreenState extends State<ClubsScreen> with BackNavigatorMixin, Error
                   mainAxisAlignment: MainAxisAlignment.center,
                   children: [
                     ClubCard(
-                      onClick: viewModel.onCardClicked,
+                      onClick: () => viewModel.onCardClicked(viewModel.runningClub),
```

Comment:
I would create 2 function onRunningCardClicked & onCyclingCardClicked. And make the onCardClicked private